### DISCUSSION TITLE
Sakurako’nun kutusu (Codeforces ödevi)

### DISCUSSION THREAD

**Discussion by iremaybuke** [sonsuzus/sonsuzus#509] [Soru Cevap]
Sakurako içinde “n” tane top olan bir kutuya sahip. Her topun kendi değeri var. Sakurako arkadaşına kutudan rastgele iki top seçmesini (iki farklı top olmak zorunda ama değerleri aynı olabilir) istiyor. Topların değerlerinin çarpımının Sakurako’nun tahmin ettiği sayıyla aynı olacağı konusunda bir iddiaya girdiler. 

Sakurako, tahmin edebileceği en iyi sayının “beklenen değer” olduğunu biliyor ama bunu nasıl hesaplayacağını unuttu. Sakurako'ya beklenen değerin ne olduğunu bulmasında yardım et.

ANA NOKTALAR

Sakurako, toplamda n top bulunduran bir kutuda, bu topların değerleri ile olasılık hesaplaması yapmaktadır.
Arkadaşının seçtiği iki top arasındaki değerlerin çarpımının beklenen değerini bilmek istemektedir.
Her test için, toplam test sayısı t ile birlikte, her testte yer alan topların sayısı n ve bu topların değerleri verilen bir diziden oluşmaktadır.
Beklenen değer hesaplaması, iki elemanın kombinasyonları üzerinden gerçekleştirilecektir.
Topların değerleri 0 ile 10^9 arasında değişebilir.
Çözüme ulaşırken, toplam n değeri tüm test vakaları için 2⋅10^5'i geçmeyecektir.
Mod işlemi, sonuçları daha yönetilebilir bir biçimde ifade etmeye yardımcı olmaktadır.

GİRDİ
3
3
3 2 3
4
2 2 2 4
5
1 2 3 4 5

ÇIKTI
7
6
500000012

Not
İlk test için, Sakurako'nun arkadaşı bu toplar çiftini seçebilir: (a1,a2), (a1,a2), (a2,a3). Değerleri sırasıyla 3.2=6, 3.3=9, 3.2=6 yani beklenen değer 6+9+6 / 3 =7.

**Comment by sonsuzus** [comment]
Soru bu kadar mı? Örnek girdi çıktıları veya formülü vs ? Çeviri yetersiz gibi duruyor soruyu anlamak imkansız bu hali ile.
